grafana/grafana
# Elastic: Add `id` field to Elastic responses to allow permalinking

**What is this feature?**

Basically this PR just copies the `_id` field to a `id` field in Elastic responses. This allows permalinking in Grafana and is compliant to the Logs dataplane.

**How to test**

1. Make sure the `enableElasticsearchBackendQuerying` feature toggle is not set to `false`
2. Query ES
3. Notice log lines have a link-button now
4. Ensure `id` field is present in the resulting DataFrame

**References:**

- https://github.com/grafana/grafana/issues/67379
- https://github.com/grafana/grafana/pull/71654; we decided to do this change now because it would allow a quick win (permalinks in ES), but not much effort migrating, since it's the `id` field anyways.
